tests/test_shards.py

```diff
@@ -125,6 +125,30 @@ def prepare_shards_test(monkeypatch: pytest.MonkeyPatch):
     },
 }
 
+# This package depends on the
```

Comment:
We could generate package(1-n), perhaps 100, and give each package a probability to depend on earlier packages in the series. More distant packages representing less likely dependencies. By using a fixed random number seed we could generate a predictable synthetic dependency graph.